tests/test_mega_hardware_serial/test_mega_hardware_serial.ino

```diff
@@ -0,0 +1,122 @@
+// This test suite is built for Arduino MEGA versions only;
+// it tests with servos attached to one serial port
+// (Serial1 is the hard-coded default) and will report 
+// test results on Serial.
+
+// Hardware Requirements:
+// - Arduino MEGA, MEGA ADK
+// - AX-12 connected directly to Serial1 port, 
+//   without tristate buffer
+
+// - By C. Matlack 2017-05-08
+
+
+#include <ArduinoUnit.h>
+#include <Dynamixel.h>
+#include <DynamixelInterface.h>
+#include <DynamixelMotor.h>
+#include <SoftwareSerial.h>               
+
+const unsigned int id = 2;
+const long unsigned int baudrate = 1000000;
+
+HardwareDynamixelInterface interface(Serial1);
+DynamixelDevice device(interface, id);
+DynamixelMotor motor(interface, id);
+
+
+void setup() {
+  Serial.begin(115200);
+}
+
+//------------ Custom Assertions ----------------------------
+// Custom Assertion for memory leak in function
+void assertNoLeak(void (*callback)())
+{
+  int start = freeMemory();
+  (*callback)();
+  int end = freeMemory();
+  assertEqual(0, (start - end));
+}
+
+// Custom Assertion for comparing byte arrays
+void assertBytesEqual(const uint8_t* expected, const uint8_t* actual, int size) {
+  for (int i = 0; i < size; i++) {
+    // Serial.print("> ");Serial.print(expected[i]);Serial.print(" ");Serial.println(actual[i]);
+    if (expected[i] != actual[i]) {
+      Serial.print("\nassertBytesEqual() failing at index ");
+      Serial.println(i);
+    }
+    assertEqual(expected[i], actual[i]);
+  }
+}
+
```

Comment:
They are not required by the current tests. Recommend leaving them in for future utility, but fine removing them as well.